lib/HALTarget/IMU.cpp

```diff
@@ -0,0 +1,226 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  IMU implementation
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IMU.h"
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool IMU::init()
+{
+    bool isInitSuccessful = m_imu.init();
+    if (isInitSuccessful)
+    {
+        m_imu.enableDefault();
+        // TD074    Make sure that a Full Scale of 245 dps of the gyros are enough
+        // TD075	Make sure that it is valid to make the Full Scale setting in IMU::init()
+        switch (m_imu.getType())
+        {
+        case Zumo32U4IMUType::LSM303D_L3GD20H:
+            // 0x00  means +/- 245 dps according to L3GD20H data sheet
+            m_imu.writeReg(L3GD20H_ADDR, L3GD20H_REG_CTRL4, 0x00);
+            break;
+        case Zumo32U4IMUType::LSM6DS33_LIS3MDL:
+            // 0x00  means +/- 245 dps according to LSM6DS33 data sheet
+            m_imu.writeReg(LSM6DS33_ADDR, LSM6DS33_REG_CTRL2_G, 0x00);
+            break;
+        default:
+            break;
```

Comment:
If there is a unknown type, is the initialization successful?

Reply:
From the documentation of Zumo32U4IMU::init():
Returns True if the sensor type was detected succesfully; false otherwise.

So this case is covered by the built-in function.